Assets/Scripts/EOSSessionsManager.cs

```diff
@@ -1862,5 +2021,419 @@ public void DeclineLobbyInvite()
+                SocketId = socketId
+            };
+
+            Result result = EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().AcceptConnection(ref options);
+
+            if (result != Result.Success)
+            {
+                Debug.LogErrorFormat("EOSSessionsManager (OnIncomingSessionsConnectionRequest): Eror while accepting connection, code: {0}", result);
+            }
+            else
+            {
+                Log($"EOSSessionsManager (OnIncomingSessionsConnectionRequest): Successfully accepted connection from {options.RemoteUserId} on socket {P2PSessionStatusSocketName}");
+            }
+        }
+
+        /// <summary>
+        /// Upon a user that is connected disconnecting, this method closes out the connection.
+        /// </summary>
+        /// <param name="data">Data, containing the product user id of the user to disconnect from.</param>
+        private void OnIncomingSessionsDisconnect(ref OnRemoteConnectionClosedInfo data)
+        {
+            SocketId socketId = new SocketId()
+            {
+                SocketName = P2PSessionStatusSocketName
+            };
+
+            CloseConnectionOptions closeOptions = new CloseConnectionOptions()
+            {
+                LocalUserId = EOSManager.Instance.GetProductUserId(),
+                RemoteUserId = data.RemoteUserId,
+                SocketId = socketId
+            };
+
+            Result result = EOSManager.Instance.GetEOSPlatformInterface().GetP2PInterface().CloseConnection(ref closeOptions);
+
+            if (result != Result.Success)
+            {
+                Debug.LogErrorFormat("EOSSessionsManager (OnIncomingSessionsDisconnect): Eror while closing connection, code: {0}", result);
+            }
+            else
+            {
+                Log($"EOSSessionsManager (OnIncomingSessionsDisconnect): Successfully closed connection with {closeOptions.RemoteUserId} on socket {P2PSessionStatusSocketName}");
+            }
+        }
+        
+        /// <summary>
+        /// Utility function to send a message to the owner of a session.
+        /// </summary>
+        /// <param name="localSessionName">The local name for the session.</param>
+        /// <param name="messageDetail">The message detail to send. This message detail is used to inform the owner of what action to take.</param>
+        private void InformSessionOwnerWithMessage(string localSessionName, string messageDetail)
+        {
+            // Find the session with this name
+            // Identify the owner of the session
+            // Send them a packet informing them of joining status
+
+            Session localSession = GetSession(localSessionName);
+
+            if (localSession == null)
+            {
+                Debug.LogErrorFormat("EOSSessionsManager (InformSessionOwnerWithMessage): No local session with name \"{0}\" was found.", localSessionName);
+                return;
+            }
```

Comment:
Mayyyyyybe... Here, it is an error. We're expecting that you should be in a session with this localSessionName. Not every time that we try to get a session does it result in an error.

That said I do generally prefer the TryGet model; it avoids null comparisons. Which isn't a big deal for the Session object, but is a big deal for Unity capital O Objects. Looking at the `GetSession` function, which existed before this PR, it kind of hilarious is just a wrapper around something actually doing TryGetValue. Since I'm the only consumer of this function, I'm comfortable changing it over... or adding another one? This is a place where the API of the sample would be changing, but nothing _in_ our samples currently relied on this.

Reply:
I've implemented the TryGetSession, but I'm not sold on putting the comment inside there.

Reply:
That's a fair observation. In similar circumstances, I have opted to include a `boolean` parameter in the `TryGet` function that determines whether an error is thrown. Typically I default the `boolean` parameter's value to whichever is the most common use-case.